ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,166 @@
+    LOG.info('prefix=%s | sufix=%s | kwargs=%s' %
+             (prefix, sufix, str(kwargs.items())))
+
+    metric_dic = {}
+    extract_unit = kwargs.get('extract_unit')
+    special_label = kwargs.get('special_label')
+    for entry in payload:
+        if special_label == 'fan':
+            e = re.sub(r'[\d].*$', '', entry.lower())
+            e = re.sub(r'[\(\)]', '', e).split()
+            label = '_'.join(e)
+        else:
+            e = re.sub(r"[\d]+", "", entry).lower().split()
+            label = '_'.join(e[:-1]).replace('-', '_')
+
+        if extract_unit:
+            sensor_read = payload[entry]['Sensor Reading'].split()
+            if len(sensor_read) > 1:
+                sufix = '_' + sensor_read[-1].lower()
+
+        if special_label == 'memory':
+            if 'mem' not in label and 'memory' not in label:
+                label = 'memory_' + label
+
+        metric_name = re.sub(r'[\W]', '_', prefix + label + sufix)
+        if metric_name[0].isdigit():
+            metric_name = metric_name.lstrip('0123456789')
+        if metric_name in metric_dic:
+            metric_dic[metric_name].append(entry)
+        else:
+            metric_dic[metric_name] = [entry]
+    return metric_dic
+
+
+def extract_labels(entries, payload, node_name):
+    """ This function extract the labels to be used by a metric
+
+    If a metric has many entries we add the 'Sensor ID' information as label
+    otherwise we will only use the default label that is the 'node_name' and
+    'Entity ID'.
+
+    e.g: for Temperature we have two entries for baremetal_temperature_celsius
+    metric ('Temp (0x1)' and 'Temp (0x2)') and one entry for 'Inlet Temp (0x5)'
+    and other for 'Exhaust Temp (0x6)', this will produce a dictionary where
+    the keys are the entries and the values are the respective label to be used
+    when writting the metrics in the Prometheus format.
+    {'Inlet Temp (0x5)': '{node_name=...}',
+     'Exhaust Temp (0x6)': '{node_name=...}',
+     'Temp (0x1)': '{node_name=...,sensor=Temp1}',
+     'Temp (0x2)': '{node_name=...,sensor=Temp2}'}
+
+    returns: a dictionarty of dictionaries {<entry>: {label_name: label_value}}
+    """
+    LOG.info('extract_labels function called with: entries=%s | payload=%s | \
+             node_name=%s' % (str(entries), str(payload), node_name))
+    if len(entries) == 1:
+        labels = {'node_name': node_name,
+                  'entity_id': payload[entries[0]]['Entity ID']}
+        return {entries[0]: labels}
+    entries_labels = {}
+    for entry in entries:
+        try:
+            entity_id = payload[entry]['Entity ID']
+            sensor_id = payload[entry]['Sensor ID']
+            metric_label = {'node_name': node_name,
+                            'entity_id': entity_id,
+                            'sensor_id': sensor_id}
+            entries_labels[entry] = metric_label
+        except Exception as e:
+            LOG.exception(e)
```

Comment:
For `LOG.exception` you need to provide context, the exception itself will be logged anyway. So something like
```python
LOG.exception("Failed to process entry %s", entry)
```
The same applies to where I commented about s/error/exception/ above